web-access-control/protected-operation/read-access-bob.feature

```diff
@@ -1,119 +1,210 @@
 Feature: Only Bob can read (and only that) a resource when granted read access
   # Grant a specific agent (setAgentAccess):
-  # - full access to the parent container (to ensure the tests are specific to the resource)
-  # - restricted access to the test resources
+  # - restricted access or no access to the parent container
+  # - restricted access to the test resources, or inherited access for fictive resources
   Background: Create test resources with correct access modes
-    * def authHeaders = (method, url, public) => !public ? clients.bob.getAuthHeaders(method, url) : {}
-    * def createResources =
+    * def authHeaders =
+    """
+      function (method, url, agent) {
+        const agentLowerCase = agent.toLowerCase()
+        return agentLowerCase != 'public' ? clients[agentLowerCase].getAuthHeaders(method, url) : {}
+      }
+    """
+    * def resourcePermissions =
     """
       function (modes) {
+        if (modes && modes != 'inherited' && modes != 'no') {
+          return Object.entries({ R: 'read', W: 'write', A: 'append', C: 'control' })
+            .filter(([mode, permission]) => modes.includes(mode))
+            .map(([mode, permission]) => permission)
+        }
+        return undefined
+      }
+    """
+    * def resourceEntry =
+    """
+      function (container, type) {
+        switch (type) {
+          case 'plain':
+            return container.createResource('.txt', 'Hello', 'text/plain')
+          case 'fictive':
+            return container.reserveResource('.txt')
+          case 'rdf':
+            return container.createResource('.ttl', karate.readAsString('../fixtures/example.ttl'), 'text/turtle')
+          case 'container':
+            return container.createContainer()
+          default:
+            return undefined
+        }
+      }
+    """
+    * def createResource =
+    """
+      function (containerModes, resourceModes, resourceType, agent) {
+
+        const agentWebId = webIds.bob
+        const testContainerPermissions = resourcePermissions(containerModes == 'all' ? 'RWAC' : containerModes)
```

Comment:
You don't appear to every set `all` - is that left over from something else? It would be good to simplify this if it is not needed.

Reply:
It was a leftover from the original tests, that had all permissions applied on the container when the focus was on testing resource permissions. But at least with the CSS, it seems fine to not assign all permissions on container in those cases, so I decided not to use `all` as a result.